inbc-program/tests/unit/test_source_app_parser.py

```diff
@@ -42,26 +112,58 @@ def test_create_add_manifest_successfully(self, m_connect):
                    '</repos><filename>intel-gpu-jammy.list</filename></repo></add></applicationSource></manifest>'
         self.assertEqual(p.func(p), expected)
 
-    def test_parse_remove_arguments_successfully(self):
+    def test_create_add_minus_gpg_manifest_successfully(self):
+        p = self.arg_parser.parse_args(
+            ['source', 'application', 'add',
+             '--sources', 'deb http://example.com/ focal main restricted universe',
+             'deb-src http://example.com/ focal-security main',
+             '--filename', 'intel-gpu-jammy.list'])
+        with patch('inbm_lib.mqttclient.mqtt.mqtt.Client.connect') as m_connect:
+            Inbc(p, 'source', False)
+        expected = '<?xml version="1.0" encoding="utf-8"?><manifest><type>source</type><applicationSource>' \
+                   '<add><repo><repos>' \
+                   '<source_pkg>deb http://example.com/ focal main restricted universe</source_pkg>' \
+                   '<source_pkg>deb-src http://example.com/ focal-security main</source_pkg>' \
+                   '</repos><filename>intel-gpu-jammy.list</filename></repo></add></applicationSource></manifest>'
```

Comment:
**Maintainability**
**Suggestion:** Consolidate the repeated XML manifest structure creation into a helper function to avoid code duplication and improve test maintainability.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        def create_expected_manifest(add_or_remove, repo_details):
            # ... implementation using an XML library ...
            return tostring(manifest).decode()
        expected = create_expected_manifest('add', repo_details)
```